### PR TITLE
fix(grpc-web): missing trailers when empty resp body

### LINKED ISSUE #12202: bug: grpc-web unexpected error "upstream grpc status not received"
### Current Behavior

Hi,

I am in the process of replacing Envoy with Apisix for grpc-web request transformation. My tests show that many things seem to work already, but I experienced a bug with the following edge case:
I have a server side grpc streaming response that **immediately** returns an error when called. This results in **two** calls of the error handler function on the frontend. The first error contains the following unexpected error message: `upstream grpc status not received`
The second error contains the expected error message from the server.
With Envoy only the expected server error is received on the frontend.

### Expected Behavior

Only the expected server error is delivered to the frontend.

### Error Logs

_No response_

### Steps to Reproduce

I created a minimal example to reproduce this edge case [here](https://github.com/Tereius/apisix_bug_minimal_example):
[Here](https://github.com/Tereius/apisix_bug_minimal_example/blob/c39a4cfe2f5116bf4d5239001407830aaa7ca5a5/cmd/main.go#L19) you find the immediate error response on the server side.
[Here](https://github.com/Tereius/apisix_bug_minimal_example/blob/c39a4cfe2f5116bf4d5239001407830aaa7ca5a5/frontend/script.js#L34) you find the frontend error handler that gets called twice.

### Environment

- APISIX version (run `apisix version`): 3.12.0
- Operating system (run `uname -a`): Linux 6.14.3-zen1-1-zen
- OpenResty / Nginx version (run `openresty -V` or `nginx -V`): openresty/1.27.1.1
- etcd version, if relevant (run `curl http://127.0.0.1:9090/v1/server_info`): -
- APISIX Dashboard version, if relevant: -
- Plugin runner version, for issues related to plugin runners: -
- LuaRocks version, for installation issues (run `luarocks --version`): -

### PR DESCRIPTION
**Pull request by bzp2010**
### Description

Fixes #12202

The problem that this PR solves is fully described in the comments of the code.

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: apisix/plugins/grpc-web.lua [hunk 1]
```diff
@@ -185,6 +185,48 @@ function _M.header_filter(conf, ctx)
...
+    --   ref: https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-WEB.md
+    --
+    -- Therefore, when this situation exists, we confirm here that the `grpc-status` response header
+    -- has been sent and set a flag.
+    -- In subsequent body_filter phases, if it finds that this flag has been set, it skips attempts
+    -- to append additional trailer encoding to the end of the response body.
+    -- This is compliant, i.e., this is a trailers-only response, send the gRPC metadata contained
+    -- in the trailers in the HTTP response header and ensure that an empty response body is
+    -- sent (no trailers block in the end of response body).
+    ctx.grpc_web_metadata_sent = ngx.var.sent_http_grpc_status ~= nil
```
Comment (bzp2010) [inline apisix/plugins/grpc-web.lua]:
```suggestion
    ctx.grpc_web_metadata_sent = ctx.var.sent_http_grpc_status ~= nil
```

**Comment by SkyeYoung** [comment]
Need to fix the CI.

**Comment by Revolyssup** [comment]
waiting on CI pass for review

**Review by Revolyssup — APPROVED** [review]
